x-pack/plugin/otel-data/src/main/resources/component-templates/semconv-resource-to-ecs@mappings.yaml

```diff
@@ -0,0 +1,133 @@
+              process.runtime.version:
+                type: keyword
+                ignore_above: 1024
+              os.name:
+                type: keyword
+                ignore_above: 1024
+              os.type:
+                type: keyword
+                ignore_above: 1024
+              os.description:
+                type: keyword
+                ignore_above: 1024
+              os.version:
+                type: keyword
+                ignore_above: 1024
+              k8s.deployment.name:
+                type: keyword
+                ignore_above: 1024
+              k8s.namespace.name:
+                type: keyword
+                ignore_above: 1024
+              k8s.node.name:
+                type: keyword
+                ignore_above: 1024
+              k8s.pod.name:
+                type: keyword
+                ignore_above: 1024
+              k8s.pod.uid:
+                type: keyword
+                ignore_above: 1024
+      service.node.name:
+        type: alias
+        path: resource.attributes.service.instance.id
+      service.environment:
+        type: alias
+        path: resource.attributes.deployment.environment
+      cloud.service.name:
+        type: alias
+        path: resource.attributes.cloud.platform
+      container.image.tag:
+        type: alias
+        path: resource.attributes.container.image.tags
+      host.architecture:
+        type: alias
+        path: resource.attributes.host.arch
+      process.executable:
+        type: alias
+        path: resource.attributes.process.executable.path
+      service.runtime.name:
+        type: alias
+        path: resource.attributes.process.runtime.path
+      service.runtime.version:
+        type: alias
+        path: resource.attributes.process.runtime.version
+      host.os.name:
+        type: alias
+        path: resource.attributes.os.name
+      host.os.platform:
+        type: alias
+        path: resource.attributes.os.type
+      host.os.full:
+        type: alias
+        path: resource.attributes.os.description
+      host.os.version:
+        type: alias
+        path: resource.attributes.os.version
+      kubernetes.deployment.name:
+        type: alias
+        path: resource.attributes.k8s.deployment.name
+      kubernetes.namespace:
+        type: alias
+        path: resource.attributes.k8s.namespace.name
+      kubernetes.node.name:
+        type: alias
+        path: resource.attributes.k8s.node.name
+      kubernetes.pod.name:
+        type: alias
+        path: resource.attributes.k8s.pod.name
+      kubernetes.pod.uid:
+        type: alias
+        path: resource.attributes.k8s.pod.uid
+# Below is a non-ECS field, ideally we should have this in a separate file
+# I'd imagine having 3 files:
+# 1) for the fixed mapping for known resource.attributes.* fields that we want to alias
+# 2) this file that aliases ECS -> SemConv
+# 3) a new file that aliases non-ECS fields the UI uses -> SemConv
```

Comment:
The alias fields need to be defined in the same component template as the fields that they target. That's because each component template is validated individually (in addition to the result of the composition). I don't think that we need to be super strict about splitting ECS and not technically ECS fields. Technically speaking, the `kubernetes.*` fields are also not part of ECS, yet I don't think it makes much sense to have a separate file for it.

Bottom line is that I think it's fine the way it is.